### ISSUE TITLE
Crash when override existing preset

### ISSUE THREAD

**Issue by OCGDFS** [FAForever/fa#1543]
I made a preset for training/sandbox games. When i try to change the map and override the existing preset, the game crashes.
[log.txt](https://github.com/FAForever/fa/files/490170/log.txt)

**Comment by Uveso** [comment]
Confirmed,
i tryed to save a preset under a new name and the game stops working.
Sound is still running but screen is freezed. Looks like an endless loop.

And there is no errormessage inside game.log.

**Comment by aeoncleanse** [comment]
Someone git bisect and find what the hell broke this...

**Comment by Uveso** [comment]
This is maybe since playerdata is using lazy vars:
https://github.com/FAForever/fa/commit/64f86aa110a10074c2e8cb14316e2b8be32f4436

There are now some functions inside the playeroptions that can't saved as array to game.prefs.

Its crashing in `prefs.lua` inside function `SetToCurrentProfile` while executing `SetPreference('profile', profile)`

I dont know how to revert this into a normal array, so i can't fix it :/

**Comment by aeoncleanse** [comment]
```ben-k@DESKTOP-N4IVHU8 MINGW64 /d/GITHUB/fa ((d2d1dde...)|BISECTING)
$ git bisect bad
d2d1ddeb129d37c955c12a68efb9d019fb08aec9 is the first bad commit
commit d2d1ddeb129d37c955c12a68efb9d019fb08aec9
Author: Michel Jung <michel.jung89@gmail.com>
Date:   Sun Jan 17 02:36:06 2016 +0100

    Add rehost functionality

:040000 040000 9e89f2543b16fd6bcd28d892598fee028bd89559 4d6858501dd29dfa28d010cc3ba1e9d27dfcc7ef M       loc
:040000 040000 a61ac37b9e777df72f5096e18bd72d484bf642ee 142d71094b7aecd0185edf3a328af2c3d9498e10 M       lua

ben-k@DESKTOP-N4IVHU8 MINGW64 /d/GITHUB/fa ((d2d1dde...)|BISECTING)
```

This is apparently the thing that broke it.

**Comment by Uveso** [comment]
Is it possible to load a savegame from another player ?

If yes, i have a savegame here to reproduce the error.
Save game starts at gametime 05:00
Game is crashing at 05:34

http://faforever.uveso.de/CrashIn0530.SCFASaveGame.rar

Crashes with gameversion 3659, beta and develop.

**Comment by Crotalus** [comment]
@Uveso You sure you added this to the right issue?

**Comment by Crotalus** [comment]
btw, what game version (tag, branch, hash) when you saved the game?

**Comment by Uveso** [comment]
Thank's Crotalus, its of course the wrong issue...

**Comment by aeoncleanse** [comment]
ac9e85890cd634bcbfe1e2e9d467c7bf1c2c20b9

**Comment by micheljung** [comment]
Dito: `Fix #1543` already created the reference, both ways.
